probe-rs/src/architecture/xtensa/mod.rs

```diff
@@ -204,7 +255,18 @@ impl<'probe> CoreInterface for Xtensa<'probe> {
     fn status(&mut self) -> Result<CoreStatus, Error> {
         let status = if self.core_halted()? {
             let debug_cause = self.interface.read_register::<DebugCause>()?;
-            CoreStatus::Halted(debug_cause.halt_reason())
+            let reason = if debug_cause.halt_reason()
+                == HaltReason::Breakpoint(BreakpointCause::Software)
+                && (debug_cause.break_instruction() || debug_cause.break_n_instruction())
+            {
```

Comment:
```suggestion
            let reason =
                if debug_cause.halt_reason() == HaltReason::Breakpoint(BreakpointCause::Software) {
```

As @bugadani commented (https://github.com/probe-rs/probe-rs/pull/2303#discussion_r1535472124) this was not needed. Also verified that it works after this suggestion